addons/product/pricelist.py

```diff
@@ -453,23 +297,48 @@ def _check_margin(self, cr, uid, ids, context=None):
+            elif record.product_tmpl_id:
+                res[record.id]['name'] = record.product_tmpl_id.name
+            elif record.product_id:
+                product_name = self.pool['product.product'].name_get(cr, uid, [record.product_id.id], context=context)[0][1]
+                product_name = product_name.replace('[%s]' % record.product_id.code, '')
+                res[record.id]['name'] = product_name
+            else:
+                res[record.id]['name'] = "All Products"
+            res[record.id]['price_name'] = self._compute_pricelist_name(cr, uid, record.id, name, args, context=context)
+        return res
+
     _columns = {
-        'name': fields.char('Rule Name', help="Explicit rule name for this pricelist line."),
-        'price_version_id': fields.many2one('product.pricelist.version', 'Price List Version', required=True, select=True, ondelete='cascade'),
+        'name': fields.function(_price_list_item_name, type="char", string='Name', store=True, multi='price_item', help="Explicit rule name for this pricelist line."),
         'product_tmpl_id': fields.many2one('product.template', 'Product Template', ondelete='cascade', help="Specify a template if this rule only applies to one product template. Keep empty otherwise."),
         'product_id': fields.many2one('product.product', 'Product', ondelete='cascade', help="Specify a product if this rule only applies to one product. Keep empty otherwise."),
         'categ_id': fields.many2one('product.category', 'Product Category', ondelete='cascade', help="Specify a product category if this rule only applies to products belonging to this category or its children categories. Keep empty otherwise."),
-        'min_quantity': fields.integer('Min. Quantity', required=True,
+        'min_quantity': fields.integer('Min. Quantity',
             help="For the rule to apply, bought/sold quantity must be greater "
               "than or equal to the minimum quantity specified in this field.\n"
               "Expressed in the default unit of measure of the product."
             ),
         'sequence': fields.integer('Sequence', required=True, help="Gives the order in which the pricelist items will be checked. The evaluation gives highest priority to lowest sequence and stops as soon as a matching item is found."),
-        'base': fields.selection(_price_field_get, 'Based on', required=True, 
-                                    size=-1, # here use size=-1 to store the values as integers
-                                    help='Base price for computation. \n Public Price: The base price will be the Sale/public Price. \n Vendor Section on Product or Cost Price : The base price will be the vendor price if it is set, otherwise it will be the cost price. \n Other Pricelist : Computation of the base price based on another Pricelist.'),
+        'base': fields.selection([('1', 'Public Price'), ('2', 'Cost'), ('-1', 'Other Pricelist')], string="Based on", required=True, size=-1,
```

Comment:
Please use real names for keys, instead of numbers:
  1 -> list_price
  2 -> standard_price
  -1 -> pricelist